#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                    		Type      Level		Description                                                           			Default    Min   Max
#Navigation settings
gen.add("goal1_pos_x",          		double_t, 	0,      "x coordinate of goal1 position.",                            					2.0, -100.0, 100.0)
gen.add("goal1_pos_y",          		double_t, 	0,      "y coordinate of goal1 position.",                                  			-3.3, -100.0, 100.0)
gen.add("goal1_pos_z",          		double_t, 	0,      "z coordinate of goal1 position.",                              				0.0, -100.0, 100.0)
gen.add("goal1_orientation",    		double_t, 	0,      "goal1 orientation as radian for z axis.",                                      0.0, 0.0, 6.28)

gen.add("goal2_pos_x",          		double_t, 	0,      "x coordinate of goal2 position.",                            					0.4, -100.0, 100.0)
gen.add("goal2_pos_y",          		double_t, 	0,      "y coordinate of goal2 position.",                                  			-0.95, -100.0, 100.0)
gen.add("goal2_pos_z",          		double_t, 	0,      "z coordinate of goal2 position.",                              				0.0, -100.0, 100.0)
gen.add("goal2_orientation",    		double_t, 	0,      "goal2 orientation as radian for z axis.",                                      0.0, 0.0, 6.28)

#General settings for motion cue
gen.add("backOff", 						bool_t,   	0,		"Is true for BO motion cue. Is false for ST motion cue.",						True)
gen.add("wait_duration", 				int_t, 		0,		"number of secs to wait until robot continues after the motion cue",			2, 0, 60)
gen.add("motion_cue_distance", 			double_t, 	0,		"desired distance between human and robot in meters for motion cue.",			1.6, 1.0, 9.0)
gen.add("human_motion_frame_distance", 	int_t, 0,			"Approaching distance of a person from frame to frame to recognize humans.",	3, 0, 20)

#BackOff settings
gen.add("back1_velocity", 				double_t, 	0,		"Velocity of first backwards movement in m/s. Can be between -0.7 and 0.",		-0.3, -0.7, 0.0)
gen.add("forw_velocity", 				double_t, 	0,		"Velocity of the forward movement in m/s. Can be between 0 and 0.7.",			0.5, 0.0, 0.7)
gen.add("back2_velocity", 				double_t, 	0,		"Velocity of the second backwards movement in m/s. Can be between -0.7 and 0.",	-0.3, -0.7, 0.0)
gen.add("back1_duration", 				double_t, 	0,		"Duration of first backwards movement in seconds.",								1.0, 0.0, 10.0)
gen.add("forw_duration", 				double_t, 	0,		"Duration of the forward movement in seconds.",									0.5, 0.0, 10.0)
gen.add("back2_duration",   			double_t,   0,		"Duration of the second backwards movement in seconds.",						0.5, 0.0, 10.0)

exit(gen.generate("turtlebot2i_lfe_navigation", "lfe_nav", "LfeNavReconfigure"))
